src/test/java/com/gradle/develocity/bamboo/DevelocityPreJobActionTest.java

```diff
@@ -198,6 +234,6 @@ void addsAccessKeyToContext(String pluginKey) {
         develocityPreJobAction.execute(stageExecution, buildContext);
 
         // then
-        verify(variableContext, times(1)).addLocalVariable(Constants.ACCESS_KEY, accessKey);
+        verify(variableContext, times(1)).addLocalVariable(Constants.ACCESS_KEY, shortLivedToken);
```

Comment:
I would add an assertion somewhere to check the access key not being there

Reply:
variableContext is a mock, so we don't really set anything in the scope of this test

Reply:
I think we can have an assertion in `it.com.gradle.develocity.bamboo.injection.GradleInjectionTest` given the `MockDevelocityServer` is enriched

Reply:
We cannot assert in those tests as there the build is triggered on Bamboo and we cannot query the secrets.
Sure, we can verify the call somehow on the MockDevelocityServer, but that doesn't assert that access key is not set.

The current assertion in the mock I think is sufficient as that was the only place we set the access key value to the secrets and test verifies that we called the method with the other value.

Reply:
I meant to echo the access key in the job and then check that the value is the mock token and not the access key